h2/src/main/org/h2/table/Column.java

```diff
@@ -133,8 +146,12 @@ public int hashCode() {
         return table.getId() ^ name.hashCode();
     }
 
+    public boolean isEnumerated() {
+        return enumerators != null && enumerators.length > 0;
```

Comment:
checking for dataType.enumerated is fine. 

I think enumerators should be null for non-ENUM data types.  I'd rather have a NullPointerException immediately if code does something wrong, than have weird behaviour further down the code.